server/hotrod/src/main/java/org/infinispan/server/hotrod/tx/TxFunctions.java

```diff
@@ -0,0 +1,114 @@
+package org.infinispan.server.hotrod.tx;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import org.infinispan.commons.marshall.AdvancedExternalizer;
+import org.infinispan.server.core.ExternalizerIds;
+import org.infinispan.transaction.xa.GlobalTransaction;
+
+/**
+ * Functions and Predicated and their externalizer.
+ * <p>
+ * To be used on cache streams.
+ *
+ * @author Pedro Ruivo
+ * @since 9.1
+ */
+public class TxFunctions {
```

Comment:
Ah, when the cache has joined the replicated cache but did not receive all data yet, it probably has to execute a stream operation remotely. In any way, doing the transaction originator check using a RPC is a no-go IMO.